provider/storage_benchmark.py

```diff
@@ -0,0 +1,329 @@
+"""
+Module for providing storage benchmark tools for testing block device.
+
+Available function:
+- generate_instance: Generate a instance by specified storage benchmark class
+                     to test block io by specified benchmark tool.
+
+Available class:
+- StorageBenchmark: Define a class provides common methods to test block io.
+- Iozone: Define a class provides methods to test file I/O performance by
+          iozone benchmark.
+- Fio: Define a class provides methods to test block I/O performance by
+       fio benchmark.
+"""
+
+import logging
+import os
+import re
+
+from platform import machine
+
+from virttest import utils_misc
+
+from avocado.core import exceptions
+
+GIT_DOWNLOAD = 'git'
+CURL_DOWNLOAD = 'curl'
+
+TAR_UNPACK = 'tar'
+
+
+class StorageBenchmark(object):
+    """
+    Create a Benchmark class which provides common interface(method) for
+    using Benchmark tool to run test.
+
+    """
+    cmds = {'linux': {'_symlinks': 'ln -s -f %s %s',
+                      '_list_pid': 'pgrep -xl %s',
+                      '_kill_pid': 'killall %s',
+                      '_rm_file': 'rm -rf {}'},
+            'windows': {'_symlinks': 'mklink %s %s',
+                        '_list_pid': 'TASKLIST /FI "IMAGENAME eq %s',
+                        '_kill_pid': 'TASKKILL /F /IM %s /T',
+                        '_rm_file': 'RD /S /Q "{}"'}}
+    tar_map = {'.tar': '-xvf', '.tar.gz': '-xzf',
+               '.tar.bz2': '-xjf', '.tar.Z': '-xZf'}
+    download_cmds = {GIT_DOWNLOAD: 'rm -rf {0} && git clone {1} {0}',
+                     CURL_DOWNLOAD: 'curl -o {0} {1}'}
+    unpack_cmds = {TAR_UNPACK: '_tar_unpack_file'}
+
+    def __init__(self, os_type, session, name):
+        """
+        :param session: session connected guest
+        :param name: the name of benchmark
+        :type name: str
+        """
+        self.session = session
+        self.name = name
+        self.os_type = os_type
+        self.env_files = []
+
+    def __getattr__(self, item):
```

Comment:
for `item` not in `['_symlinks', '_list_pid', '_kill_pid', '_rm_file']`, `__getattr__` should raise `AttributeError`.

Reply:
Thanks, have updated.

Reply:
Thanks, have updated.